### DISCUSSION TITLE
Copier naming not very google-friendly compared to Cookiecutter or Yeoman

### DISCUSSION THREAD

**Discussion by adepretis** [copier-org/copier#949] [General]
Hi there,

I am a newbie to templating tools like Copier or Cookiecutter. Initially only Cookiecutter was on my radar and when doing the usual research before jumping on a specific technology-train, I discovered Copier and was convinced that for our corporate use-cases Copier seems to be the more future-proof solution with a better UX/DX. A big thanks to all authors and contributors by the way!

For a Copier newbie there's a catch though: the name/term is really very un-google-able. When searching for specific Cookiecutter (or Yeoman) problems/solutions/patterns it's quite easy to find tech articles, tutorials, and so on. With "Copier" this is really a challenge and sometimes frustrating when learning Copier details, looking for specific problems/solutions, and so on. I was a minute away to cut corners and switch to Cookiecutter - multiple times ;-)

Could this maybe hinder/slow down Copier adoption out there? What do the authors/contributors and other Copier (power)users think about this topic?

**Comment by yajo** [comment]
Hello! Thanks for your kind words, and welcome to the community!

This project's former maintainer also raised this issue in https://github.com/copier-org/copier/issues/97 but we closed it because Copier is a nice name, after all.

I didn't think about its googlability at the time. It is a valid point, but also we should keep in mind that renaming the project would mean a lot of work in terms of:
- URL redirections.
- Losing our current positioning. There are podcasts, blog posts, links around the web... that already talk about Copier and would be talking about nothing now.
- New namespace in readthedocs, pypi, github...
- People having to deal with a big breaking update (CLI/API usage would break).

It seems like it's gonna be a lot of work to fix a problem by introducing a lot of other problems, so honestly I'm not very motivated to change it now...

**Comment by adepretis** [comment]
I totally understand such a step would come with plenty of strings and effort attached. I was just wondering if further adoption would benefit from enough from such a step to justify the effort - we all know, Google and Stackoverflow/StackExchange are vital for daily business and learning new tech-stacks. I'll try ChatGPT in the future for Copier research, could be interesting ;-)

**Comment by sisp** [comment]
> For a Copier newbie there's a catch though: the name/term is really very un-google-able. When searching for specific Cookiecutter (or Yeoman) problems/solutions/patterns it's quite easy to find tech articles, tutorials, and so on. With "Copier" this is really a challenge and sometimes frustrating when learning Copier details, looking for specific problems/solutions, and so on. I was a minute away to cut corners and switch to Cookiecutter - multiple times ;-)

Isn't part of the problem that Cookiecutter has significantly more traction? There are many more templates on GitHub (whatever they're worth), more blog articles, more StackOverflow questions, etc. I'm sure those things contribute to Google's ranking algorithm. The term "cookiecutter" without further context is not very expressive IMO.

From a technical point of view, I think Copier is the clear winner over all alternatives I've seen.

I think the project name "Copier" is not bad and would become more visible with more adoption and more content on the web, which of course is a :chicken: and :egg: problem.

**Reply by adepretis** [reply]
I honestly don't know if more traction/content out there will fix this "problem". Cookiecutter has nearly no competition - search for "cookiecutter kitchen" or "cookiecutter baking", 99% of all content refers to "cookie cutter" (with space) and is only found because of the "kitchen" or "baking" context and Google's smartness ;-) ... "cookiecutter" results in tech not kitchen related content from the very beginning of the Cookiecutter journey.

Copier competes with all Brothers, Xerox, HPs, Canons and a massive load of copier-related content (manuals, reviews, price comparisons, ...). There's of course the point of user-tracking and Google's smartness, which results in more "Copier-the-template-generator" related content vs. "Copier-the-machine" content over time when you spent enough time searching for the template-context and clicking around in that bubble. So over time, the personal experience (if you embrace tracking/customized Google experience) indeed improves.

Yeoman has no competition.

**Comment by adepretis** [comment]
But in the end it's most likely not worth a lengthy discussion. I just wanted to point it out ;-)

**Comment by yajo** [comment]
I found that searching for "copier python" yields better results. However I always wanted to keep copier lang-agnostic.

💭 Yeoman is the project name whereas the command is just `yo`. We could rename the human name of the project to something still related to "copier" while still keeping copier for urls, cli and API. So that if you usually search for "hp copier" or "toshiba copier", then you can also search for "(some magic word) copier" and get here.

Copier introduces a new concept that I usually call "project (or code)  lifecycle management" (the main difference with the others). But that's too long for a magic word 🤣.

Any ideas?

PS: FWIW I don't mind having a small community 🙂

**Comment by sisp** [comment]
By the way, Copier's GitHub stars trend looks quite promising :tada:: https://star-history.com/#copier-org/copier&Date

**Reply by yajo** [reply]
Wooo cool
